src/shared/user-agent.js

```diff
@@ -35,3 +35,13 @@ export const isIOS = (
     (_navigator.userAgent.includes('Mac') && _ontouchend)
   );
 };
+
+/**
+ * Returns true when the device is a touch device such
+ * as android or iOS.
+ *
+ * @param _window {Window} - Test seam
+ */
+export const isTouchDevice = (_window = window) => {
+  return _window.matchMedia('(pointer: coarse)').matches;
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer#browser_compatibility the only browser that is theoretically supported but may not support `(pointer: coarse)` is Firefox < 64. These devices will simply report as non-touch, which is fine. The same will happen with CSS rules that use this query.

Reply:
I will add that link in a comment.. I think its valuable.